In [1]:
import requests
from bs4 import BeautifulSoup
import pandas as pd
import time
import logging
import lxml

Now that necessary packages are installed, going to parse the needed url. This URL has every retained salary transaction since 2013. So I am going to scrape the cap friendly website (Retained Salary Tab) and get this into a dataframe that I can download and put into the MySQL database

In [2]:
url_v2 = "https://www.capfriendly.com/retained-salary-transactions"

req = requests.get(url_v2)
soup = BeautifulSoup(req.content)  # make a soup of html & css from the web page

In [3]:
df = pd.read_html(url_v2, header=0, index_col = 0, na_values=["-"])[0]

In [4]:
df.shape

(206, 6)

After running df.shape, we can see our dataframe has 206 rows (players) and 6 columns (attributes about that player contract). 

Now doing the actual scraping and putting into a dataframe

In [5]:
req = requests.get(url_v2)
soup = BeautifulSoup(req.content)  # make a soup of html & css from the web page


url_start = "https://www.capfriendly.com/retained-salary-transactions"


print(f"Start scapring page 1")

time.sleep(1)  # let the page download the results
url = url_start   # we parse the needed page by adding a parameter for url
df = pd.read_html(url, header=0, index_col = 0, na_values=["-"])[0]
df = df.reset_index()  # to have player name as a separate column
print(df.shape[0], f"rows were retrieved from page number 1")

result_df = df

Start scapring page 1
206 rows were retrieved from page number 1


In [6]:
result_df.head(5)

PLAYER TEAM           DATE PERCENTAGE INITIAL CAP HIT  \
0   Jacob Markström  CGY  Jun. 19, 2024     31.25%      $6,000,000   
1     Tyler Toffoli  NJD   Mar. 8, 2024     50.00%      $4,250,000   
2  Evgeny Kuznetsov  WSH   Mar. 8, 2024     50.00%      $7,800,000   
3       Tomas Hertl  SJS   Mar. 8, 2024     17.05%      $8,137,500   
4        Jake Allen  MTL   Mar. 8, 2024     50.00%      $3,850,000   

  RETAINED CAP HIT ACTIVE CONTRACT  
0       $1,875,000               ✔  
1       $2,125,000             NaN  
2       $3,900,000               ✔  
3       $1,387,500               ✔  
4       $1,925,000               ✔

Now, I have the entire retained salary transactions table from CapFriendly. So time to export it to a csv, and then upload it into the MySQL database I created. 

In [7]:
result_df.to_csv('CapFriendly RetainedSalaryTransactions2.csv', encoding='utf-8')
result_df.to_csv('CapFriendly RetainedSalaryTransactions.csv')

The csv file above is a weird file type **Forgot to add the .csv extension above, issue is fixed now. Just have to replace the weird "✔ with a Yes as when a check mark is scraped off their website, it can not get represented correctly in excel. So just replace that symbol with the text Yes.